keel-lemur/src/main/kotlin/com/netflix/spinnaker/keel/lemur/LemurCertificateResolver.kt

```diff
@@ -0,0 +1,40 @@
+package com.netflix.spinnaker.keel.lemur
+
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ec2.ApplicationLoadBalancerSpec
+import com.netflix.spinnaker.keel.api.ec2.EC2_APPLICATION_LOAD_BALANCER_V1_2
+import com.netflix.spinnaker.keel.api.plugins.Resolver
+import kotlinx.coroutines.runBlocking
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean
+import org.springframework.stereotype.Component
+
+@Component
+@ConditionalOnBean(LemurService::class)
+class LemurCertificateResolver(
+  private val lemurService: LemurService
+) : Resolver<ApplicationLoadBalancerSpec> {
+  override val supportedKind = EC2_APPLICATION_LOAD_BALANCER_V1_2
+
+  override fun invoke(resource: Resource<ApplicationLoadBalancerSpec>): Resource<ApplicationLoadBalancerSpec> =
+    resource.copy(
+      spec = resource.spec.copy(
+        listeners = resource.spec.listeners.mapTo(mutableSetOf()) { listener ->
+          listener.certificate?.let { name ->
+            listener.copy(
+              certificate = findCurrentCertificate(name)
+            )
+          } ?: listener
+        }
+      )
+    )
+
+  private fun findCurrentCertificate(name: String) =
+    runBlocking {
+      val certificate = lemurService.certificateByName(name).items.first()
```

Comment:
I'm guessing both the list here and `certificate.replacedBy` can be empty, so perhaps use `firstOrNull()` and check the result so this doesn't throw `NoSuchElementException`?

Reply:
Yeah, I also wonder if it's potentially necessary to follow a chain longer than one link. I wasn't able to find any example in Lemur where that was the case, but just in case I'm going to add some further capabilities here.